Dockerfile

```diff
@@ -3,74 +3,100 @@
+    perfsonar-testpoint \
+    supervisor \
+    rsyslog && \
+    # clean up
+    yum clean all && \
+    rm -rf /var/cache/yum/*
 
 #
 # PostgreSQL Server
 #
-# Based on a Dockerfile at
-# https://raw.githubusercontent.com/zokeber/docker-postgresql/master/Dockerfile
-
-# Postgresql version
-ENV PG_VERSION 9.5
-ENV PGVERSION 95
-
-# Set the environment variables
-ENV PGDATA /var/lib/pgsql/9.5/data
+# shouldn't be necessary but isn't added to path via rpm
+ENV PATH="/usr/pgsql-9.5/bin:${PATH}"
+# declare database location
+ENV PGDATA="/var/lib/pgsql/9.5/data/"
 
 # Initialize the database
-RUN su - postgres -c "/usr/pgsql-9.5/bin/pg_ctl init"
-
+RUN su postgres -c 'pg_ctl init'
 # Overlay the configuration files
-COPY postgresql/postgresql.conf /var/lib/pgsql/$PG_VERSION/data/postgresql.conf
-COPY postgresql/pg_hba.conf /var/lib/pgsql/$PG_VERSION/data/pg_hba.conf
-
-# Change own user
-RUN chown -R postgres:postgres /var/lib/pgsql/$PG_VERSION/data/*
-
-# End PostgreSQL Setup
-
-
-# -----------------------------------------------------------------------------
+COPY ["postgresql/postgresql.conf", \
+      "postgresql/pg_hba.conf", \
+      "/var/lib/pgsql/9.5/data/"]
+RUN chown -R postgres:postgres $PGDATA
 
 #
 # pScheduler Database
 #
-# Initialize pscheduler database.  This needs to happen as one command
-# because each RUN happens in an interim container.
-
-COPY postgresql/pscheduler-build-database /tmp/pscheduler-build-database
-RUN  /tmp/pscheduler-build-database
-RUN  rm -f /tmp/pscheduler-build-database
-
-
-# -----------------------------------------------------------------------------
+# Initialize pscheduler database
+RUN su postgres -c "pg_ctl start -w -t 60" && \
+    # Generate the password file
+    random-string --safe --length 60 \
+    > '/etc/pscheduler/database/database-password' && \
+    # Generate the DSN file
+    printf "host=127.0.0.1 dbname=pscheduler user=pscheduler password=%s\n" \
+    cat /etc/pscheduler/database/database-password \
+    > /etc/pscheduler/database/database-dsn && \
+    # Generate a PostgreSQL password file
+    # Format is hostname:port:database:username:password
+    printf "*:*:pscheduler:pscheduler:%s\n" \
+    cat /etc/pscheduler/database/database-password \
+    > "/etc/pscheduler/database/pgpassfile" && \
+    chmod 400 /etc/pscheduler/database/pgpassfile && \
+    # Build the database
+    pscheduler internal db-update && \
+    # Set the password in the pScheduler database to match what's on the
+    # disk.
+    ( \
+        printf "ALTER ROLE pscheduler WITH UNENCRYPTED PASSWORD '" \
+        && tr -d "\n" < "/etc/pscheduler/database/database-password" \
+        && printf "';\n" \
+    ) | postgresql-load && \
+    # Shut down
+    su postgres -c "pg_ctl stop  -w -t 60"
 
 # Rsyslog
-# Note: need to modify default CentOS7 rsyslog configuration to work with Docker, 
-# as described here: http://www.projectatomic.io/blog/2014/09/running-syslog-within-a-docker-container/
```

Comment:
This is not necessary to note here because it's noted in the rsyslog files that we copy over.